examples/Azure_IoT_Central_ESP32/AzureIoT.h

```diff
@@ -0,0 +1,809 @@
+
+/*
+ * @brief        Function to disconnect and deinitialize a MQTT client.
+ * @remark       When this function is called the MQTT client instance referenced by `mqtt_client_handle` shall disconnect
+ *               from the server and any functions of the MQTT client API that destroy the instance shall be called
+ *               (so any allocated memory resources can be released).
+ *
+ * @param[in]    mqtt_client_handle    A pointer to the instance of the MQTT client previously created with
+ *                                     `mqtt_client_init_function_t` function.
+ * 
+ * @return       int                   0 on success, non-zero if any failure occurs.
+ *                                     Returning non-zero results in the Azure IoT Client going into error state.
+ */
+typedef int (*mqtt_client_deinit_function_t)(mqtt_client_handle_t mqtt_client_handle);
+
+/*
+ * @brief        Function to send an MQTT PUBLISH.
+ * @remark       When this function is invoked, the caller expects the actual MQTT client (referenced by `mqtt_client_handle`) 
+ *               to invoke the appropriate function in the MQTT client API to publish an MQTT message.
+ *
+ * @param[in]    mqtt_client_handle    A pointer to the instance of the MQTT client previously created with
+ *                                     `mqtt_client_init_function_t` function.
+ * @param[in]    mqtt_message          A structure containing the topic name, payload and QoS to be used to publish
+ *                                     an actual MQTT message.
+ * 
+ * @return       int                   The packet ID on success, or NEGATIVE if any failure occurs.
+ *                                     If the QoS in `mqtt_message` is:
+ *                                     - AT LEAST ONCE, the Azure IoT client expects `azure_iot_mqtt_client_connected` 
+ *                                       to be called once the MQTT client receives a PUBACK.
+ *                                     - AT MOST ONCE, there should be no PUBACK, so no further action is
+ *                                       needed for this PUBLISH.
+ */
+typedef int (*mqtt_client_publish_function_t)(mqtt_client_handle_t mqtt_client_handle, mqtt_message_t* mqtt_message);
+
+/*
+ * @brief        Function to send an MQTT SUBSCRIBE.
+ * @remark       When this function is invoked, Azure IoT client expects the actual MQTT client (referenced by
+ *               `mqtt_client_handle`) to invoke the appropriate function in the MQTT client API to subscribe to 
+ *               an MQTT topic.
+ *
+ * @param[in]    topic           The string containing the complete MQTT topic name to subscribed to.
+ *                               This string is always NULL-terminated.
+ * @param[in]    topic_length    The length of `topic`. It can be used in case the MQTT client takes a 
+ *                               topic name as a non-NULL-terminated string.
```

Comment:
nit: "in case" makes it sound optional
```suggestion
 * @param[in]    topic_length    The length of `topic`.
```

Reply:
It is optional. 
User can chose a MQTT client that works with ptr+size or null-terminated strings for topic names.